### ISSUE TITLE
Feature regression: Sender's Name & Email Address

### ISSUE THREAD

**Issue by kunalsood** [leafac/kill-the-newsletter#102]
As of recently, the author tags on all item in all my feeds look like :-
```
<author>
  <name>Kill the Newsletter!</name>
  <email>kill-the-newsletter@leafac.com</email>
</author>
```
Is there any chance we can go back to reporting the real sender's name & email address?

**Comment by leafac** [comment]
Hi @kunalsood,

Thanks for reaching out.

I changed this recently when I was simplifying some things in Kill the Newsletter!

It’s something we can bring back, yes, but it’s annoying to migrate data, so I’ll wait for more people to ask for this before I do the work.

Thanks.

**Comment by leafac** [comment]
I ended up implemeting this in version 2.0.6 because there were some other data migrations to do.

**Comment by kunalsood** [comment]
Hi @leafac,

Thanks!
I really do appreciate the time & effort you put into this.

I do have a concern regarding the new implementation though.

My Morning Brew newsletter feed now has author tags that look something like the following :-
```
<author>
  <name>delivery_20240616072314.35722233.3442996@bounce.morningbrew.com</name>
  <email>delivery_20240616072314.35722233.3442996@bounce.morningbrew.com</email>
</author>
```
or
```
<author>
  <name>delivery_20240612044635.35684490.116214@bounce.morningbrew.com</name>
  <email>delivery_20240612044635.35684490.116214@bounce.morningbrew.com</email>
</author>
```

I'm curious where these email addresses are coming from? I ask because as far as I'm aware, Moring Brew only send emails from email addresses that look like `crew@morningbrew.com` or `crew@community.morningbrew.com` with sender name being `Morning Brew`.

I'm seeing a similar issue with other newsletters as well. For example, my feed of TLDR Newsletter now contains author tags that look like:- 

```
<author>
  <name>01000190164f1d9f-4d74239e-ddf6-4049-ad01-bdaf1484bd10-000000@dailyupdate.tldrnewsletter.com</name>
  <email>01000190164f1d9f-4d74239e-ddf6-4049-ad01-bdaf1484bd10-000000@dailyupdate.tldrnewsletter.com</email>
</author>
```

While in my email, these newsletters come from `dan@tldrnewsletter.com` with sender name being `TLDR`.

Could these author tags be containing the reply-to emails & not the actual sender's info?

**Comment by leafac** [comment]
> Could these author tags be containing the reply-to emails & not the actual sender's info?

Yes, you got it. Right now we’re picking up the email address from the `FROM` field in the SMTP message envelope. That’s the most reliable “source of truth,” but perhaps it isn’t the most user-friendly piece of data.

I’ll look into this…

**Comment by kunalsood** [comment]
@leafac Thanks for your efforts!

Out of curiosity, where were you getting sender's info from prior to sometime around May 5, 2024?
I ask because the author tags used to have the correct (i.e. more user-friendly piece of data) info till about May 5, 2024, at least on my feeds in your hosted instance at kill-the-newsletter.com.

Until that point of time, IIRC, the author tags used to look something like these:-
```
<author>
TLDR <dan@tldrnewsletter.com>
</author>
```
```
<author>
Morning Brew <crew@morningbrew.com>
</author>
```

**Comment by leafac** [comment]
I think we used to get that information from the message `From` field, as opposed to the `FROM` in the envelope of the SMTP exchange. Perhaps we should just revert to that…
